### PR TITLE
Internal: Editor V2- Page Indication + Settings [ED-9664]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: packages/packages/documents/src/init.ts [hunk 1]
```diff
@@ -1,19 +1,28 @@
...
-	filler: PrimaryAction,
-} );
+function registerTopBarMenuItems() {
+	injectIntoCanvasDisplay( {
+		name: 'documents-canvas-display',
+		filler: CanvasDisplay,
+	} );
+
+	injectIntoPrimaryAction( {
+		name: 'top-bar-primary-action',
```
Comment (StyleShit) [inline packages/packages/documents/src/init.ts:25]:
```suggestion
		name: 'primary-action',
```

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts [hunk 2]
```diff
@@ -66,9 +67,15 @@ describe( '@elementor/documents - Sync Store', () => {
...
-				type: 'wp-page',
 				title: 'Document 2',
-				status: 'publish',
+				type: {
+					value: 'wp-page',
+					label: 'wp-page',
+				},
+				status: {
+					value: 'publish',
+					label: 'publish',
```
Comment (StyleShit) [inline packages/packages/documents/src/sync/__tests__/sync-store.test.ts:77]:
```suggestion
					label: 'Publish',
```

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts [hunk 3]
```diff
@@ -101,12 +108,17 @@ describe( '@elementor/documents - Sync Store', () => {
...
 			title: 'Document 2',
-			status: 'publish',
-			type: 'wp-page',
+			type: {
+				value: 'wp-page',
+				label: 'wp-page',
+			},
+			status: {
+				value: 'publish',
+				label: 'publish',
```
Comment (StyleShit) [inline packages/packages/documents/src/sync/__tests__/sync-store.test.ts:120]:
```suggestion
				label: 'Publish',
```

File: packages/packages/documents/src/sync/sync-store.ts [hunk 4]
```diff
@@ -11,39 +11,39 @@ import {
...
+				entities: Object.entries( documentsManager.documents )
+					.reduce( ( acc: Record<string, Document>, [ id, document ] ) => {
+						acc[ id ] = normalizeV1Document( document );
 
-			dispatch( setDocuments( normalizedDocuments ) );
+						return acc;
+					}, {} ),
+				hostId: documentsManager.getInitialId(),
+				activeId: documentsManager.getCurrentId(),
+			} ) );
```
Comment (StyleShit) [inline packages/packages/documents/src/sync/sync-store.ts:37]: use variable

File: packages/packages/v1-adapters/src/hooks/__tests__/test-utils.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { EditMode, ExtendedWindow } from '../../readers/types';
+
+const extendedWindow = window as unknown as ExtendedWindow;
+
+export function mockIsRouteActive( implementation: ( route: string ) => boolean ) {
+	extendedWindow.$e = {
+		routes: {
+			isPartOf: jest.fn( implementation ),
+		},
+	};
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/hooks/__tests__/test-utils.ts:10]: cleanup global vars 
also others

File: packages/packages/v1-adapters/src/hooks/__tests__/use-is-preview-mode.test.ts [hunk 6]
```diff
@@ -44,11 +37,12 @@ describe( '@elementor/v1-adapters - useIsPreviewMode', () => {
...
-			mockedGetCurrentEditMode.mockReturnValue( 'edit' );
+			mockGetCurrentEditMode( () => 'edit' );
 			dispatchEditModeChange();
 		} );
 
 		// Assert.
 		expect( result.current ).toBe( false );
 	} );
 } );
+
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/hooks/__tests__/use-is-preview-mode.test.ts:48]: remove empty line

File: packages/packages/v1-adapters/src/hooks/__tests__/use-is-route-active.test.ts [hunk 7]
```diff
@@ -98,3 +91,4 @@ describe( '@elementor/v1-adapters - useIsRouteActive', () => {
 		expect( result.current ).toBe( false );
 	} );
 } );
+
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/hooks/__tests__/use-is-route-active.test.ts:94]: same as above

File: packages/packages/v1-adapters/src/hooks/index.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
 export { default as useIsPreviewMode } from './use-is-preview-mode';
 export { default as useIsRouteActive } from './use-is-route-active';
+export { default as useRouteStatus } from './use-route-status';
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/hooks/index.ts:3]: A-Z

File: packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts [hunk 9]
```diff
@@ -348,6 +313,26 @@ describe( '@elementor/v1-adapters/listeners', () => {
 		delete extendedWindow.__elementorEditorV1LoadingPromise;
 	} );
 
+	it( 'should not listen to event if isReady equals to `false`', () => {
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts:316]:
```suggestion
	it( 'should not trigger callback when the application is not ready', () => {
```

File: packages/packages/v1-adapters/src/listeners/is-ready.ts [hunk 10]
```diff
@@ -0,0 +1,9 @@
+let isReady = false;
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/listeners/is-ready.ts:1]: explain file

File: packages/packages/v1-adapters/src/listeners/is-ready.ts [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+
+export function getIsReady() {
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/listeners/is-ready.ts:3]:
```suggestion
export function isReady() {
```

File: packages/packages/v1-adapters/src/readers/types.ts [hunk 12]
```diff
@@ -1,3 +1,5 @@
+export type EditMode = 'edit' | 'preview' | 'picker'
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/readers/types.ts:1]:
```suggestion
export type EditMode = 'edit' | 'preview' | 'picker';
```

File: packages/tests/setup.ts [hunk 13]
```diff
@@ -14,5 +16,9 @@ beforeEach( () => {
 	jest.mocked( console.info ).mockImplementation( () => null );
 	/* eslint-enable no-console */
 
-	resetInjections();
+	flushInjections();
```
Comment (StyleShit) [inline packages/tests/setup.ts:19]: maybe 
```
delete window.$e;
delete window.elementor;
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: core/documents-manager.php [hunk 14]
```diff
@@ -524,6 +527,15 @@ public function ajax_save( $request ) {
...
 		];
 
+		$post_status_object = get_post_status_object( $main_post->post_status );
+
+		if ( $post_status_object ) {
+			$return_data['config']['document']['status'] = [
+				'value' => $post_status_object->name,
+				'label' => $post_status_object->label,
+			];
+		}
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply core/documents-manager.php:537]: Cant, this logic already exists

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+
+jest.mock<typeof import( '../../../hooks/use-active-document' )>( '../../../hooks/use-active-document', () => ( {
+	__esModule: true,
+	default: jest.fn( () => null ),
+} ) );
+
+jest.mock<typeof import( '../../../hooks/use-host-document' )>( '../../../hooks/use-host-document', () => ( {
+	__esModule: true,
+	default: jest.fn( () => null ),
+} ) );
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:14]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+describe( '@elementor/documents - top bar canvas display', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:16]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+	beforeEach( () => {
+		jest.mocked( useActiveDocument ).mockImplementation( () =>
+			createMockDocument( { id: 1, title: 'Active Document' } )
+		);
+		jest.mocked( useHostDocument ).mockImplementation( () =>
+			createMockDocument( { id: 2, title: 'Host Document' } )
+		);
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:23]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+	it( 'should show the title of the active document without the status', () => {
+		// Act.
+		const { queryByText } = render( <CanvasDisplay /> );
+
+		// Assert.
+		expect( queryByText( 'Active Document' ) ).toBeTruthy();
+		expect( queryByText( '(publish)' ) ).not.toBeTruthy();
+	} );
+
+	it( 'should show the title of the active document with the status', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:35]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 19]
```diff
@@ -0,0 +1,100 @@
...
+		);
+
+		// Act.
+		const { queryByText } = render( <CanvasDisplay /> );
+
+		// Assert.
+		expect( queryByText( 'Host Document' ) ).toBeTruthy();
+	} );
+
+	it( 'should not show nothing if there is no documents', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:88]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/settings-button.test.tsx [hunk 20]
```diff
@@ -0,0 +1,44 @@
+import SettingsButton from '../settings-button';
+import { openRoute, useRouteStatus } from '@elementor/v1-adapters';
+import { render } from '@testing-library/react';
+
+jest.mock( '@elementor/v1-adapters', () => ( {
+	openRoute: jest.fn(),
+	useRouteStatus: jest.fn( () => ( { isActive: false, isBlocked: false } ) ),
+} ) );
+
+describe( '@elementor/documents - Top bar settings button', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/settings-button.test.tsx:10]: No

File: packages/packages/documents/src/components/top-bar/canvas-display.tsx [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+		if ( activeDocument && activeDocument.type.value !== 'kit' ) {
+			return activeDocument;
+		}
+
+		if ( hostDocument ) {
+			return hostDocument;
+		}
+
+		return null;
+	}, [ activeDocument, hostDocument ] );
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/canvas-display.tsx:22]: Done

File: packages/packages/documents/src/components/top-bar/settings-button.tsx [hunk 22]
```diff
@@ -0,0 +1,31 @@
...
+
+	return (
+		<Tooltip title={ title }>
+			<ToggleButton
+				value="document-settings"
+				selected={ isActive }
+				disabled={ isBlocked }
+				onChange={ () => openRoute( 'panel/page-settings/settings' ) }
+			>
+				<SvgIcon viewBox="0 0 21 20">
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/settings-button.tsx:25]: Done

File: packages/packages/documents/src/hooks/__tests__/use-host-document.test.tsx [hunk 23]
```diff
@@ -0,0 +1,49 @@
...
+		} ) );
+
+		// Act.
+		const { result } = renderHookWithStore( useHostDocument, store );
+
+		// Assert.
+		expect( result.current ).toBeNull();
+	} );
+} );
+
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/hooks/__tests__/use-host-document.test.tsx:49]: Done

File: packages/packages/documents/src/hooks/index.ts [hunk 24]
```diff
@@ -1,2 +1,3 @@
 export { default as useActiveDocument } from './use-active-document';
+export { default as useHostDocument } from './use-host-document';
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/hooks/index.ts:2]: Done

File: packages/packages/documents/src/store/index.ts [hunk 25]
```diff
@@ -1,48 +1,74 @@
...
-	activeId: 0,
+	activeId: null,
+	hostId: null,
+};
+
+const createActiveDocumentReducer = (
+	callback: ( document: Document ) => void
+) => {
+	return ( state: State ) => {
+		if ( ! state.activeId ) {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/index.ts:20]: Typescript issue

File: packages/packages/documents/src/init.ts (hunk 1 above)
Comment (Nevoss) [reply packages/packages/documents/src/init.ts:25]: Done

File: packages/packages/documents/src/store/selectors.ts [hunk 26]
```diff
@@ -3,9 +3,30 @@ import { createSelector, SliceState } from '@elementor/store';
...
+
+export const selectHostDocument = createSelector(
+	selectEntities,
+	selectHostId,
+	( entities, hostId ) => {
+		if ( ! hostId ) {
+			return null;
+		}
+
+		return entities[ hostId ];
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/selectors.ts:30]: Done

File: packages/packages/documents/src/store/selectors.ts [hunk 27]
```diff
@@ -3,9 +3,30 @@ import { createSelector, SliceState } from '@elementor/store';
...
-	selectDocumentsSlice,
-	( documents ) => documents.entities[ documents.activeId ] || null,
+	selectEntities,
+	selectActiveId,
+	( entities, activeId ) => {
+		if ( ! activeId ) {
+			return null;
+		}
+
+		return entities[ activeId ];
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/selectors.ts:18]: Done

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts [hunk 28]
```diff
@@ -54,9 +49,15 @@ describe( '@elementor/documents - Sync Store', () => {
...
-				type: 'wp-page',
 				title: 'Document 1',
-				status: 'publish',
+				type: {
+					value: 'wp-page',
+					label: 'wp-page',
+				},
+				status: {
+					value: 'publish',
+					label: 'publish',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/sync/__tests__/sync-store.test.ts:59]: Have some issues with the test, lets talk about it

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts (hunk 2 above)
Comment (Nevoss) [reply packages/packages/documents/src/sync/__tests__/sync-store.test.ts:77]: Same as above

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts (hunk 3 above)
Comment (Nevoss) [reply packages/packages/documents/src/sync/__tests__/sync-store.test.ts:120]: Again

File: packages/packages/documents/src/sync/sync-store.ts (hunk 4 above)
Comment (Nevoss) [reply packages/packages/documents/src/sync/sync-store.ts:37]: Done

File: packages/packages/v1-adapters/src/hooks/__tests__/test-utils.ts (hunk 5 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/hooks/__tests__/test-utils.ts:10]: Done

File: packages/packages/v1-adapters/src/hooks/__tests__/use-is-preview-mode.test.ts (hunk 6 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/hooks/__tests__/use-is-preview-mode.test.ts:48]: Done

File: packages/packages/v1-adapters/src/hooks/__tests__/use-is-route-active.test.ts (hunk 7 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/hooks/__tests__/use-is-route-active.test.ts:94]: Done

File: packages/packages/v1-adapters/src/hooks/index.ts (hunk 8 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/hooks/index.ts:3]: Done

File: packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts (hunk 9 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/listeners/__tests__/index.test.ts:316]: Done

File: packages/packages/v1-adapters/src/listeners/is-ready.ts (hunk 10 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/listeners/is-ready.ts:1]: Done

File: packages/packages/v1-adapters/src/listeners/is-ready.ts (hunk 11 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/listeners/is-ready.ts:3]: Done

File: packages/packages/v1-adapters/src/readers/types.ts (hunk 12 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/readers/types.ts:1]: Done

File: packages/tests/setup.ts (hunk 13 above)
Comment (Nevoss) [reply packages/tests/setup.ts:19]: Done

File: packages/tests/setup.ts [hunk 29]
```diff
@@ -1,10 +1,14 @@
...
+import { flushInjections } from '@elementor/locations';
+import { flushListeners, setReady } from '@elementor/v1-adapters';
+import { deleteStore } from '@elementor/store';
 
 // Add JSDOM matchers.
 import '@testing-library/jest-dom';
 
 import '@wordpress/jest-console';
 
+let windowOriginalProps: string[];
```
Comment (StyleShit) [inline packages/tests/setup.ts:10]:
```suggestion
let windowOriginalProps: PropertyKey[];
```

File: packages/tests/setup.ts [hunk 30]
```diff
@@ -14,5 +18,20 @@ beforeEach( () => {
...
+	setReady( true );
+
+	windowOriginalProps = Object.keys( window );
+} );
+
+afterEach( () => {
+	// Delete all the props that were added to the window.
+	Object.keys( window ).forEach( ( key ) => {
+		if ( ! windowOriginalProps.includes( key ) ) {
+			delete ( window as unknown as {[key: string]: unknown} )[ key ];
```
Comment (StyleShit) [inline packages/tests/setup.ts:34]:
```suggestion
			delete ( window as unknown as Record<PropertyKey, unknown> )[ key ];
```

File: packages/packages/v1-adapters/src/listeners/is-ready.ts [hunk 31]
```diff
@@ -0,0 +1,14 @@
+/**
+ * This file is used to store the state of the isReady variable.
+ * which is used to determine if the adapter is ready to receive events (editor v1 and v2 are loaded).
```
Comment (StyleShit) [inline packages/packages/v1-adapters/src/listeners/is-ready.ts:3]:
```suggestion
 * This file is used to store the state of the isReady variable, which is used to determine 
 * if the adapter is ready to receive events (editor v1 and v2 are loaded).
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 32]
```diff
@@ -0,0 +1,101 @@
...
+		jest.mocked( useActiveDocument ).mockImplementation( () =>
+			createMockDocument( { id: 1, title: 'Active Document' } )
+		);
+
+		jest.mocked( useHostDocument ).mockImplementation( () =>
+			createMockDocument( { id: 2, title: 'Host Document' } )
+		);
+	} );
+
+	it( 'should show the title of the active document without the status (for non-publish status)', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:27]: Done

File: packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx [hunk 33]
```diff
@@ -0,0 +1,101 @@
...
+		// Act.
+		const { queryByText } = render( <CanvasDisplay /> );
+
+		// Assert.
+		expect( queryByText( 'Active Document' ) ).toBeTruthy();
+		expect( queryByText( '(publish)' ) ).not.toBeTruthy();
+	} );
+
+	it( 'should show the title of the active document with the status', () => {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/__tests__/canvas-display.test.tsx:36]: Done

File: packages/packages/documents/src/components/top-bar/canvas-display.tsx [hunk 34]
```diff
@@ -0,0 +1,25 @@
...
+import { useActiveDocument } from '../../hooks';
+import useHostDocument from '../../hooks/use-host-document';
+import SettingsButton from './settings-button';
+import Indicator from './indicator';
+
+export default function CanvasDisplay() {
+	const activeDocument = useActiveDocument();
+	const hostDocument = useHostDocument();
+
+	const document = activeDocument && activeDocument?.type?.value !== 'kit'
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/canvas-display.tsx:11]: No, if there is no active document, it should show the host

File: packages/packages/documents/src/icons/cog.tsx [hunk 35]
```diff
@@ -0,0 +1,9 @@
+import { SvgIcon } from '@elementor/ui';
+
+export default function Cog() {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/icons/cog.tsx:3]: Done

File: packages/packages/documents/src/store/index.ts [hunk 36]
```diff
@@ -1,49 +1,74 @@
...
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/index.ts:20]: Done

File: packages/tests/setup.ts (hunk 29 above)
Comment (Nevoss) [reply packages/tests/setup.ts:10]: Done

File: packages/tests/setup.ts (hunk 30 above)
Comment (Nevoss) [reply packages/tests/setup.ts:34]: Done

File: packages/packages/v1-adapters/src/listeners/is-ready.ts (hunk 31 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/listeners/is-ready.ts:3]: Done

**Review by StyleShit — DISMISSED** [review]

File: packages/packages/v1-adapters/src/hooks/use-route-status.ts [hunk 37]
```diff
@@ -0,0 +1,32 @@
...
+	blockOnKitRoutes?: boolean,
+	blockOnPreviewMode?: boolean,
+}
+
+export default function useRouteStatus(
+	route: RouteEventDescriptor['name'],
+	{
+		blockOnKitRoutes = true,
+		blockOnPreviewMode = true,
+	}: Options = {}
```
Comment (rotemee) [inline packages/packages/v1-adapters/src/hooks/use-route-status.ts:15]: Why not just one type for all?

File: packages/packages/documents/src/__tests__/test-utils.ts [hunk 38]
```diff
@@ -1,15 +1,30 @@
...
+			label: 'Published',
+		},
+		type: type || {
+			value: 'wp-page',
+			label: 'Page',
+		},
+		isDirty,
+		isSaving,
+		isSavingDraft,
+		userCan: userCan || {
```
Comment (rotemee) [inline packages/packages/documents/src/__tests__/test-utils.ts:27]: Maybe move the default with all the reset at the function arguments?

File: packages/packages/documents/src/components/top-bar/settings-button.tsx [hunk 39]
```diff
@@ -0,0 +1,30 @@
...
+
+type Props = {
+	type: Document['type']
+}
+
+export default function SettingsButton( { type }: Props ) {
+	const { isActive, isBlocked } = useRouteStatus( 'panel/page-settings' );
+
+	/* translators: %s: Post type label. */
+	const title = __( '%s settings', 'elementor' )
```
Comment (rotemee) [inline packages/packages/documents/src/components/top-bar/settings-button.tsx:15]: This is the product requirement? always putting the word "settings" at the end? a bit weird..

**Review by rotemee — CHANGES_REQUESTED** [review]

File: packages/packages/documents/src/components/top-bar/settings-button.tsx [hunk 40]
```diff
@@ -0,0 +1,30 @@
...
+		.replace( '%s', type.label );
+
+	return (
+		<Tooltip title={ title }>
+			<ToggleButton
+				value="document-settings"
+				selected={ isActive }
+				disabled={ isBlocked }
+				onChange={ () => openRoute( 'panel/page-settings/settings' ) }
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/settings-button.tsx:24]: As a see at the docs ToggleButton uses onChange

File: packages/packages/documents/src/store/index.ts [hunk 41]
```diff
@@ -1,49 +1,74 @@
...
 		name: 'documents',
 		initialState,
 		reducers: {
-			setDocuments( state, action: PayloadAction<State['entities']> ) {
-				state.entities = action.payload;
+			init( state, { payload } : PayloadAction<{
+				entities: State['entities'],
+				hostId: State['hostId'],
+				activeId: State['activeId'] }>
+			) {
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/index.ts:37]: Done

File: packages/packages/documents/src/store/index.ts [hunk 42]
```diff
@@ -1,49 +1,74 @@
...
 
-			endSaving( state ) {
+				state.entities[ state.activeId ] = action.payload;
 				state.entities[ state.activeId ].isSaving = false;
 			},
 
-			startSavingDraft( state ) {
-				state.entities[ state.activeId ].isSavingDraft = true;
-			},
+			startSavingDraft: createActiveDocumentReducer( ( document ) => document.isSavingDraft = true ),
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/store/index.ts:59]: Lets talk about it

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts [hunk 43]
```diff
@@ -54,9 +42,15 @@ describe( '@elementor/documents - Sync Store', () => {
 		expect( storeState.documents.entities ).toEqual( {
 			1: {
 				id: 1,
-				type: 'wp-page',
 				title: 'Document 1',
-				status: 'publish',
+				type: {
+					value: 'wp-page',
+					label: 'wp-page',
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/sync/__tests__/sync-store.test.ts]: Done

File: packages/packages/documents/src/sync/__tests__/sync-store.test.ts [hunk 44]
```diff
@@ -252,21 +236,55 @@ describe( '@elementor/documents - Sync Store', () => {
...
+		// Assert.
+		expect( selectActiveDocument( store.getState() )?.title ).toBe( 'test' );
+		expect( selectActiveDocument( store.getState() )?.status.value ).toBe( 'draft' );
+
+		// Act.
+		dispatchCommandAfter( 'document/save/save' );
+
+		// Assert.
+		expect( selectActiveDocument( store.getState() )?.title ).toBe( 'test title changed' );
+		expect( selectActiveDocument( store.getState() )?.status.value ).toBe( 'publish' );
```
[the remark this answers is not in the document]
Comment (Nevoss) [reply packages/packages/documents/src/sync/__tests__/sync-store.test.ts:287]: I just added the `?` and dont want to change all those tests

File: packages/packages/v1-adapters/src/hooks/use-route-status.ts (hunk 37 above)
Comment (Nevoss) [reply packages/packages/v1-adapters/src/hooks/use-route-status.ts:15]: ?, there is type for the options and one for the first argument

File: packages/packages/documents/src/__tests__/test-utils.ts (hunk 38 above)
Comment (Nevoss) [reply packages/packages/documents/src/__tests__/test-utils.ts:27]: Done

File: packages/packages/documents/src/components/top-bar/settings-button.tsx (hunk 39 above)
Comment (Nevoss) [reply packages/packages/documents/src/components/top-bar/settings-button.tsx:15]: Yeah it is

**Review by rotemee — APPROVED** [review]

**Pull request merged**